ms2query/app_helpers.py

```diff
@@ -397,19 +400,38 @@ def get_library_matches(documents_query: List[SpectrumDocument],
         def_show_topn = len(documents_library)
     cols = st.beta_columns([1, 4])
     with cols[0]:
-        show_topn = int(st.text_input("Show top n matches",
-                                      value=def_show_topn))
+        show_topn = int(
+            st.text_input("Show top n matches", value=def_show_topn))
+
     documents_library_class = DocumentsLibrary(documents_library)
     found_matches_s2v = cached_library_matching(
         documents_query, documents_library_class, model, topn, lib_num,
         model_num)
 
-    st.write("These are the library matches for your query")
     if found_matches_s2v:
-        first_found_match = found_matches_s2v[0]
-        first_found_match = first_found_match.sort_values(
-            "s2v_score", ascending=False)
-        st.dataframe(first_found_match.iloc[:show_topn])
+        first_found_match = found_matches_s2v[0].copy()  # to maintain caching
+        do_nn_predictions = st.checkbox(
+            "Remove uncertain matches with deep learning")
+        if do_nn_predictions:
+            first_found_match = get_nn_predictions(
+                first_found_match, documents_library, documents_query)
+        # make s2v_score first column
+        df_cols = first_found_match.columns.to_list()
+        s2v_i = [i for i, col in enumerate(df_cols) if col == "s2v_score"][0]
+        s2v_sc = [df_cols.pop(s2v_i)]
+        new_cols = s2v_sc + df_cols
```

Comment:
Maybe this could go into a separate small function.
And that could also be a good point to add any metadata that we want to display in the table. I would suggest:
- parent mass
- name
- maybe: smiles

Reply:
Something like this? I didn't add the smiles as the table is already getting very big. The spectrum ID and compound name are there though so a user could search one of them and find all the needed info.

Reply:
One direct connection to a structure would be nice Joris - as for example copying the SMILES into a structure drawer immediately gives insight into the structure - whereas spectrum ID does not. And if the space is limited, maybe this could be done by a "copy smiles" button? Just brainstorming here :-)

Reply:
Yes that is very true indeed. I added the smiles now as well. What do you think, is the table annoying to view now? We could also remove the spectrum ID..

Reply:
Joris, I don't have it installed/running yet - probably something on my list for January! :-) Florian, what do you think? Sometimes functionality goes over aesthetics....

Reply:
Looks good Joris. In the future we of course want to have the molecule being displayed as mouse-over in the network plot, but until then I think it is good enough to have the smiles in the table.